zio-http/src/main/scala/zhttp/http/Request.scala

```diff
@@ -34,14 +35,29 @@ trait Request extends HeaderExtension[Request] { self =>
   /**
    * Decodes the content of request as a Chunk of Bytes
    */
-  def body: Task[Chunk[Byte]] =
-    bodyAsByteBuf.flatMap(buf => Task(Chunk.fromArray(ByteBufUtil.getBytes(buf))))
+  final def body: Task[Chunk[Byte]] =
+    bodyAsByteBuf.flatMap(buf =>
+      Task(Chunk.fromArray(ByteBufUtil.getBytes(buf))).ensuring(UIO(buf.release(buf.refCnt()))),
+    )
 
   /**
    * Decodes the content of request as string
    */
-  def bodyAsString: Task[String] =
-    bodyAsByteBuf.flatMap(buf => Task(buf.toString(charset)))
+  final def bodyAsString: Task[String] =
+    bodyAsByteBuf.flatMap(buf => Task(buf.toString(charset)).ensuring(UIO(buf.release(buf.refCnt()))))
+
```

Comment:
```suggestion
  final def bodyAsByteArray: Task[Array[Byte]] = ???
```

Let's add this method, which internally has `ensuring` finalizer. Then use `bodyAsByteArray` in `body` and `bodyAsString`.

Reply:
Done